src/rank_llm/rerank/rank_listwise_os_llm.py

```diff
@@ -174,6 +224,27 @@ def create_prompt(
                 )
         return prompt, self.get_num_tokens(prompt)
 
+    def create_prompt_batched(
+        self, results: List[Result], rank_start: int, rank_end: int,
+        batch_size: int = 32
+    ) -> List[Tuple[str, int]]:
+        def chunks(lst, n):
+            """Yield successive n-sized chunks from lst."""
+            for i in range(0, len(lst), n):
+                yield lst[i:i + n]
+
+        all_completed_prompts = []
+
+        with ThreadPoolExecutor() as executor:
+            for batch in tqdm(chunks(results, batch_size), desc="Processing batches"):
+                futures = [
+                    executor.submit(self.create_prompt, result, rank_start, rank_end)
+                    for result in batch
+                ]
+                completed_prompts = [future.result() for future in as_completed(futures)]
```

Comment:
IIUC, as completed returns the futures in order of completion rather than the original input or other. Should we use executor.map here?

Reply:
good catch, check now?